pennylane/estimator/templates/select.py

```diff
@@ -356,20 +407,30 @@ def controlled_resource_decomp(
         cswap = resource_rep(qre.CSWAP)
         gate_list.append(GateCount(cswap, 4 * num_orb))
 
+        # Calculate the number of Givens rotation blocks
+        if batched_rotations is None:
+            batched_rotations = num_orb - 1
```

Comment:
So, what batched_rotations is? Is it the number of Givens rotation blocks? Please be careful with the comments, names etc.